### PR TITLE
Improve error handling in main

### PR DESCRIPTION
**Pull request by vaqxai**
Add error when hotkey not found/cannot be loaded
Return Result from main

### CODE REVIEW

File: miner/src/main.rs [hunk 1]
```diff
@@ -335,13 +335,21 @@ async fn main() {
         &*config::HOTKEY_NAME,
     );
 
-    let account_id = load_key_account_id(&hotkey_location).unwrap();
+    let account_id = match load_key_account_id(&hotkey_location) {
+        Ok(account_id) => account_id,
+        Err(e) => {
+            eprintln!("Error loading hotkey! Please verify that it exists! ({:?})", e);
+            return Err(anyhow::anyhow!(e));
```
Comment (MsRandom) [inline miner/src/main.rs]: Do not return an error. Simply `panic!` with that error message for better logging.

**Review by MsRandom — CHANGES_REQUESTED** [review]

File: miner/src/main.rs (hunk 1 above)
Comment (vaqxai) [reply miner/src/main.rs]: I disagree. The error provides the needed information and is a cleaner way of exiting than a panic. This is a user error and not a cause for a panic, it is completely recoverable too.

Comment (MsRandom) [reply miner/src/main.rs]: Returning a Result from main simply exits with a non-zero exit code and an exit message, similarly to a panic(which does the same + extra stuff like a backtrace, at the cost of having worse formatting).

The current design always panics on startup errors. If we were to merge this, it should mean changing everything to delegate the errors back to main and return them, which is out of scope for this PR. It should be changed to match the rest of the repository, and the entire repository can be changed afterwards in a separate PR to use `Result<(), E>` for handling startup errors.

(Additionally, the way you're handling it logs the error twice.)

Comment (vaqxai) [reply miner/src/main.rs]: The double report was actually intentional, as not to obscure additional info from the end user but I agree that returning `Result` from main would require all error handling to be done this way. I'll change it for a more descriptive panic for now.

**Review by MsRandom — APPROVED** [review]

**Pull request merged**